src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.Utf8JsonStreamReader.cs

```diff
@@ -0,0 +1,1801 @@
+                                    if (jsonReader.Read())
+                                    {
+                                        dependencyVersionValue = jsonReader.GetString();
+                                    }
+                                }
+                                else
+                                {
+                                    jsonReader.Skip();
+                                }
+                            }
+                        }
+
+                        VersionRange dependencyVersionRange = null;
+
+                        if (!string.IsNullOrEmpty(dependencyVersionValue))
+                        {
+                            try
+                            {
+                                dependencyVersionRange = VersionRange.Parse(dependencyVersionValue);
+                            }
+                            catch (Exception ex)
+                            {
+                                throw FileFormatException.Create(
+                                    ex,
+                                    packageSpecPath);
+                            }
+                        }
+
+                        if (dependencyVersionRange == null)
+                        {
+                            throw FileFormatException.Create(
+                                new ArgumentException(Strings.MissingVersionOnDependency),
+                                packageSpecPath);
+                        }
+
+                        // the dependency flags are: Include flags - Exclude flags
+                        var includeFlags = dependencyIncludeFlagsValue & ~dependencyExcludeFlagsValue;
+                        var libraryDependency = new LibraryDependency()
+                        {
+                            LibraryRange = new LibraryRange()
+                            {
+                                Name = propertyName,
+                                TypeConstraint = LibraryDependencyTarget.Package,
+                                VersionRange = dependencyVersionRange
+                            },
+
+                            IncludeType = includeFlags,
+                            SuppressParent = suppressParentFlagsValue,
+                            VersionCentrallyManaged = true,
+                            ReferenceType = LibraryDependencyReferenceType.Transitive
+                        };
+
+                        results.Add(libraryDependency);
+                    }
+                }
+            }
+        }
+
+        private static void ReadDependencies(
+            ref Utf8JsonStreamReader jsonReader,
+            IList<LibraryDependency> results,
+            string packageSpecPath,
+            bool isGacOrFrameworkReference)
+        {
+            if (jsonReader.Read() && jsonReader.TokenType == JsonTokenType.StartObject)
+            {
+                while (jsonReader.Read() && jsonReader.TokenType == JsonTokenType.PropertyName)
```

Comment:
Like other comments, I'm not sure the error handling here provides a good API experience. If `jsonReader.Read()` returns false, doesn't that mean the file has an error?  If  read succeeds, but the token type is wrong, that's deinately an error, but no error is being reported here.